### DISCUSSION TITLE
Consuming queues and linking the right messages to the right Airflow process

### DISCUSSION THREAD

**Discussion by domien96** [apache/airflow#30619] [Q&A]
Imagine a single rabbitMQ queue and a single DAG.

I have two processes/runs running for the single DAG. Both processes trigger some external microservice and both wait for a response message. For example, process 1 uploads a video about foo, and process 2 uploads a video about bar.
It is not determined which video will be uploaded first, as the uploading is handled by this external microservice. 

How can I ensure with airflow that once a response message is received, it gets routed to the correct process. So video "foo" should route to airflow process 1, video "bar" to airflow process 2.

TL;DR: How can I correlate queue messages to the correct airfow runs?

**Comment by potiuk** [comment]
I think it's not airflow process, it's generic rabbitmq question. You should likely ask it in rabbitmq forums how to do it
